x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -78,6 +78,18 @@ public void startDeployment(TrainedModelDeploymentTask task, ActionListener<Trai
         doStartDeployment(task, listener);
     }
 
+    public ModelStats getStats(TrainedModelDeploymentTask task) {
```

Comment:
I am wondering if this should be in the `TrainedModelAllocationNodeService`? If we do not gather the initial `RoutingState` from the cluster state immediately, we should grab it from some cached value in `TrainedModelAllocationNodeService`.